gatsby-node.js

```diff
@@ -371,3 +390,71 @@ exports.onCreateWebpackConfig = ({ actions }) => {
     },
   })
 }
+
+/**
+ * Add custom field resolvers to the GraphQL schema. Allows adding new fields to types by providing field configs,
+ * or adding resolver functions to existing fields.
+ *
+ * We are using this to save the search index as a JSON object as we create here during build time.
+ *
+ * [Gatsby Node API - createResolvers]{@link https://www.gatsbyjs.org/docs/node-apis/#createResolvers}
+ *
+ * @param {function} createResolvers
+ */
+exports.createResolvers = ({ createResolvers }) => {
+  createResolvers({
+    Query: {
+      ParliamentSearchIndex: {
+        type: GraphQLJSONObject,
+        resolve(source, args, context) {
+          const siteNodes = context.nodeModel.getAllNodes({
+            type: `MarkdownRemark`,
+          })
+          const pages = context.nodeModel.getAllNodes({
+            type: `ParliamentNavigation`,
+          })
+          return createIndex(siteNodes, pages)
+        },
+      },
+    },
+  })
+}
+
+/**
+ * Creates an elasticlunr index of all the markdown and open api documents.
+ *
+ * [Gatsby Node API - createResolvers]{@link https://www.gatsbyjs.org/docs/node-apis/#createResolvers}
+ *
+ * @param {Array} nodes An array containing all the markdown documents
+ * @param {Object} pages The contents of ParliamentNavigation
+ */
+const createIndex = async (nodes, pages) => {
+  const index = elasticlunr()
+  index.setRef(`id`)
+  index.addField(`title`)
+  index.addField(`body`)
+  index.addField(`path`)
+  index.addField(`type`)
+
+  for (node of nodes) {
+    const { slug } = node.fields
+    let title = searchTree(pages, slug)
+    if (title) {
+      const doc = {
+        id: slug,
+        title: title,
+        body: node.internal.content,
+        path: slug,
+        type: "docs",
+      }
+      index.addDoc(doc)
+    }
+  }
+
+  // Open API specs are not in graphql db, hence this hack
```

Comment:
I've had luck using this plugin to add swagger/openapi info into graphql:
https://www.gatsbyjs.org/packages/gatsby-source-openapi-aggregate/?=openapi

Probably overkill for this particular PR, but worth mentioning as we continue providing better integration with those files